### PR TITLE
add SQL drivers

### PR DESCRIPTION
**Pull request by hmoffatt**
This PR enables the SQL drivers for Postgres, MySQL and Firebird. All require additional -dev packages to be installed, Firebird also required a patch to the Qt sources which I have included for all versions back to 5.9.0. Before that the Qt configure system was different.

Unfortunately it isn't possible to add a compile-time test like the other tests that exist, because plugins can't be tested for at compile time. If the build is modified to actually run the compiled test programs then the test could inspect the result of QSqlDatabase::drivers() to check that the plugins were detected.

### CODE REVIEW

**Comment by darkmattercoder** [comment]
Cool, and thank you so much for that contribution. I want to figure out a way to have that in 5.6 as well. however, it is really good to have it at least in newer versions. I will have a look into that.

Or is the patch the thing that is not required in versions below 5.9 ? And adding the -dev packages will work there as well?

**Comment by darkmattercoder** [comment]
CI fails for 5.9.0

```
+ sed -i -e s/"-lgds"/"-lfbclient"/ ../qtbase/src/plugins/sqldrivers/configure.json
sed: can't read ../qtbase/src/plugins/sqldrivers/configure.json: No such file or directory
```

**Comment by hmoffatt** [comment]
Ah, it looks like that file came in with v5.9.1, not v5.9.0. Can you 
please just remove that line from the 5.9.0 script?

https://github.com/qt/qtbase/tree/v5.9.1/src/plugins/sqldrivers

**Comment by darkmattercoder** [comment]
Are you sure, that it will work that way? 
With the additional packages and the patch, I see the following in the config output:

```
Qt Sql Drivers:
  DB2 (IBM) .............................. no
  InterBase .............................. no
  MySql .................................. no
  OCI (Oracle) ........................... no
  ODBC ................................... no
  PostgreSQL ............................. no
  SQLite2 ................................ no
  SQLite ................................. yes
    Using system provided SQLite ......... no
```

I would feel better with tests.
I configured the pipeline to run all tests after building them, see https://github.com/darkmattercoder/qt-build/commit/755da1e11ef1a82a0e2c4dec87990a6c5d372b1b and db48b08
Can you please add one test to verify that the sql drivers do their job?
When I manually test only using your suggested `QSqlDatabase::drivers()` I get a list drivers listed, no matter whether the additional dev packages are installed and the patch applied, or not......

**Comment by hmoffatt** [comment]
Thanks for updating the build process. I will develop a test. It will be a few days before I have time.

Which version is that configure output from? I only tested with 5.12.5 but it works for me:

```Qt Sql Drivers:
  DB2 (IBM) .............................. no
  InterBase .............................. yes
  MySql .................................. yes
  OCI (Oracle) ........................... no
  ODBC ................................... no
  PostgreSQL ............................. yes
  SQLite2 ................................ no
  SQLite ................................. yes
    Using system provided SQLite ......... no
  TDS (Sybase) ........................... no
```

**Comment by darkmattercoder** [comment]
The output was from one of the 5.9 versions. I do not remember the exact one though.

**Comment by hmoffatt** [comment]
I've added a test on my branch. It checks for MySQL and Postgres only currently, unless we can get IBASE (Firebird) available in all versions.

**Comment by darkmattercoder** [comment]
Looks good, as I assume Mysql, sqlite and postgres are the most common ones, I think we can live with that. Let's see what the ci build goes like. Looking forward to close #4

File: buildconfig/configure-5.9.0.sh [hunk 1]
```diff
@@ -13,6 +13,7 @@ make -j$CORE_COUNT > /dev/null 2>&1 || make -j$CORE_COUNT
 make test > /dev/null 2>&1 || make test
 make install > /dev/null 2>&1 || make install
 cd ..
+sed -i -e 's/"-lgds"/"-lfbclient"/' ../qtbase/src/plugins/sqldrivers/configure.json
```
Comment (darkmattercoder) [inline buildconfig/configure-5.9.0.sh]: That line should get removed, it's the last thing left, all other builds were fine.

**Review by darkmattercoder — APPROVED** [review]

File: buildconfig/configure-5.9.0.sh (hunk 1 above)
Comment (darkmattercoder) [inline buildconfig/configure-5.9.0.sh]:
```suggestion
```
Please remove that line, everything else looks perfekt

**Review by darkmattercoder — APPROVED** [review]

**Comment by darkmattercoder** [comment]
Thank you for your contributions!
close #4

**Pull request merged**